### PR TITLE
j2-overlord: Clean up some VAG functions

### PR DESCRIPTION
**Pull request by Ziemas**
Some cleanup of vag related functions in overlord.

### CODE REVIEW

**Comment by Ziemas** [comment]
I was about to do more here, but it's probably for the best if split stuff up in case there's regressions.

Help with testing would be appreciated.

**Comment by ManDude** [comment]
Can you also rename `CalculateFallofVolume` to `CalculateFalloffVolume` and `gCurve` to `gCurves` as per https://github.com/open-goal/jak-project/pull/3288/commits/354f3a2f81b493364f282c60aa05b888de64b689 ?

File: game/overlord/jak1/ssound.cpp [hunk 1]
```diff
@@ -115,6 +115,32 @@ void InitSound_Overlord() {
   }
 }
 
+void UpdateLocation(Sound* sound) {
+  if (sound->id == 0) {
+    return;
+  }
+
+  if (g_game_version == GameVersion::Jak1) {
```
Comment (ManDude) [inline game/overlord/jak1/ssound.cpp:123]: This can be removed now I think.

**Review by ManDude — APPROVED** [review]
Didn't notice any regressions.

**Pull request merged**